src/hotspot/share/gc/g1/g1RemSet.cpp

```diff
@@ -1193,16 +1193,9 @@ class G1MergeHeapRootsTask : public WorkerTask {
       // implicitly rebuild anything else during eager reclaim. Note that at the moment
       // (and probably never) we do not enter this path if there are other kind of
       // remembered sets for this region.
-      r->rem_set()->clear(true /* only_cardset */);
-      // Clear_locked() above sets the state to Empty. However we want to continue
-      // collecting remembered set entries for humongous regions that were not
-      // reclaimed.
-      r->rem_set()->set_state_complete();
```

Comment:
In original code, in set_state_complete() it `_state = Complete;`, but in new patch, `_state = Complete;` is not called, is this expected?

Reply:
Only remembered sets of state `Complete` remembered sets reach here.